File: 1107/median.py
class Solution:

    def median(self,nums):
        length = len(nums)
        print(length)
        if length%2==0:
            median = nums[length//2-1] + nums[length//2]
            median /= 2
        else:
            median = nums[length//2]

        return median
        
    def concat(self,nums1,nums2):
        idx1,idx2=0,0
        len1,len2=len(nums1),len(nums2)
        print(len1,len2)
        nums = []
        for _ in range(len1+len2):
            if not idx1 >= len1: n1 = nums1[idx1]
            if not idx2 >= len2: n2 = nums2[idx2]

            if idx1 >= len1 :
                nums.append(n2)
                idx2 += 1
                
            elif n1 >= n2:
                nums.append(n2)
                idx2 += 1

            elif idx2 >= len2:
                nums.append(n1)
                idx1 += 1

            elif n1 < n2:
                nums.append(n1)
                idx1 += 1
            


        return nums

    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        
        nums = self.concat(nums1,nums2)
        return self.median(nums)
        

File: 1107/median.cpp
#include <stdio.h>
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    double median(vector<int>& nums){
        double median;
        int len = nums.size();
        int center = (int)len/2;
        

        if(len%2==0){
            median = nums[center - 1] + nums[center];
            median = median/2.0;
        }else{
            median = nums[center];
        }

        return median;
    }
    vector<int> concat(vector<int>& nums1, vector<int>& nums2){
        vector<int> nums = {};
        int idx1,idx2;
        idx1 = 0;
        idx2 = 0;

        for(int i=0; i<nums1.size()+nums2.size();i++){
            if(idx1 >= nums1.size()){
                nums.insert(nums.end(),nums2[idx2]);
                idx2 += 1;
            }
            else if(idx2 >= nums2.size() || nums1[idx1] < nums2[idx2]){
                nums.insert(nums.end(),nums1[idx1]);
                idx1 += 1;
            }
            else if(nums1[idx1] >= nums2[idx2]){
                nums.insert(nums.end(),nums2[idx2]);
                idx2 += 1;
            }
            else if(nums1[idx1] < nums2[idx2]){
                nums.insert(nums.end(),nums1[idx1]);
                idx1 += 1;
            }
        }

        return nums;
    }

    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int> nums = concat(nums1,nums2);
        return median(nums);
    }
};



int main (){
    Solution s;
    vector<int> nums1 = {10000};
    vector<int> nums2 = {10001};
    double median = s.findMedianSortedArrays(nums1,nums2);
    cout << median << endl;
    return 0;
}